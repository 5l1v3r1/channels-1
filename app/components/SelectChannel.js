// @flow
import React from 'react';
import styles from './SelectChannel.scss';

const SelectChannel = ({
  toggleChannelList
}: {
  toggleChannelList: () => void
}) => (
  <div className={styles.selectChannel}>
    <div className={styles.prompt}>
      <p className={styles.message}>
        No Active Channel
        <span>{"Well don't just sit there..."}</span>
      </p>
      <button className={styles.select} onClick={toggleChannelList}>
        Select a Channel
      </button>
    </div>
  </div>
);

export default SelectChannel;
