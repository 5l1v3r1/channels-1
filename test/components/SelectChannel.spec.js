import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SelectChannel from '../../app/components/SelectChannel';

function setup() {
  const props = {
    toggleChannelList: jest.fn()
  };

  const enzymeWrapper = shallow(<SelectChannel {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('SelectChannel', () => {
  it('should render self', () => {
    const { enzymeWrapper } = setup();
    const tree = toJson(enzymeWrapper);

    expect(tree).toMatchSnapshot();
  });

  it('should call toggleChannelList', () => {
    const { props, enzymeWrapper } = setup();
    const select = enzymeWrapper.find('.select');

    select.simulate('click');

    expect(props.toggleChannelList).toBeCalled();
  });
});
