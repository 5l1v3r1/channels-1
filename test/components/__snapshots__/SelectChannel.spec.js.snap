// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectChannel should render self 1`] = `
<div
  className="selectChannel"
>
  <div
    className="prompt"
  >
    <p
      className="message"
    >
      No Active Channel
      <span>
        Well don't just sit there...
      </span>
    </p>
    <button
      className="select"
      onClick={[Function]}
    >
      Select a Channel
    </button>
  </div>
</div>
`;
